redis/spring-cloud-starter-stream-sink-redis/src/main/java/org/springframework/cloud/stream/app/redis/sink/RedisSinkProperties.java

```diff
@@ -79,51 +79,63 @@ public void setTopic(String topic) {
 		this.topic = topic;
 	}
 
-	public Expression getKeyExpression() {
-		return key != null ? new LiteralExpression(key) : parser.parseExpression(keyExpression);
+	public Expression keyExpression() {
```

Comment:
Like Ilaya said, we need exact getter/setter/field tuple for boot metadata generation to work properly.
This is another method with added logic (I could have named it even more differently)